blockchain/blockchain.go

```diff
@@ -1149,3 +1144,51 @@ func (b *Blockchain) Finalise(pending *Pending, sign BlockSignFunc) error {
 		return storeStateUpdate(txn, pending.Block.Number, pending.StateUpdate)
 	})
 }
+
+func (b *Blockchain) GenesisState() (*core.StateUpdate, error) {
+	var genesisStateUpdate *core.StateUpdate
+	return genesisStateUpdate, b.database.View(func(txn db.Transaction) error {
+		var err error
+		genesisStateUpdate, err = genesisState(txn)
+		return err
+	})
+}
+
+func genesisState(txn db.Transaction) (*core.StateUpdate, error) {
+	var genesisStateUpdate *core.StateUpdate
+	return genesisStateUpdate, txn.Get(db.GenesisState.Key(), func(b []byte) error {
+		genesisStateUpdate = new(core.StateUpdate)
+		return encoder.Unmarshal(b, genesisStateUpdate)
+	})
+}
+
+func (b *Blockchain) InitGenesisState(stateDiff *core.StateDiff, classes map[felt.Felt]core.Class) error {
+	return b.database.Update(func(txn db.Transaction) error {
+		_, err := genesisState(txn)
+		if !errors.Is(err, db.ErrKeyNotFound) {
+			return errors.New("genesis state already initiliazed")
+		}
+
+		state := core.NewState(txn)
+		if err = state.Update(0, stateDiff, classes); err != nil {
```

Comment:
This will log changes to history for block zero, correct? Do we want that?

Reply:
Yeah, it is not ideal, but doesn't really hurt either. Since it requires some changes to `core.State` API, I opted to deal with it later if needed.